OpenSim/Simulation/Model/Geometry.h

```diff
@@ -0,0 +1,659 @@
+ * Geometry items to represent itself in graphics window. The relation between
+ * a ModelComponent and specific Geometry utilizes the Component mechanism, as
+ * the specific pieces of geometry are treated as subcomponents. The placement
+ * of the Geometry in 3D spae is computed from the Frame that the Geometry is
+ * "Connected" to.
+ *
+ * Geometry (and all its subclasses) serve as the set of higher level primitives
+ * available to OpenSim component writers to express the Geometry of interest.
+ * The Geometry class handles serialization and also the translation to a set of
+ * DecorativeGeometry objects that gets passed to the Visualization system to be
+ * rendered. 
+ *
+ * @author Ayman Habib
+ */
+
+// Base Geometry
+class OSIMSIMULATION_API Geometry : public Component {
+    OpenSim_DECLARE_ABSTRACT_OBJECT(Geometry, Component);
+public:
+    // Properties common to all Geometry types are included as Properties
+    // of the base class.
+    // Scale factors
+    OpenSim_DECLARE_PROPERTY(scale_factors, SimTK::Vec3,
+        "Scale factors in X, Y, Z directions respectively.");
+    // Default display properiies e.g. Representation, color, texture, etc.
+    OpenSim_DECLARE_UNNAMED_PROPERTY(Appearance,
+        "Default appearance for this Geometry");
+
+    enum DisplayPreference {
+        Hide = 0,       ///<Hide geometry from display
+        DrawPoints = 1, ///< Use a cloud of points.
+        DrawWireframe = 2, ///< Use a line drawing.
+        DrawSurface = 3, ///< Use a shaded surface.
+
+        DrawDefault = -1  ///< Let someone else decide.
+    };
+    //--------------------------------------------------------------------------
+    // CONSTRUCTION
+    //--------------------------------------------------------------------------
+public:
+    // Default constructor, does nothing
+    Geometry()
+    {
+        setNull();
+        constructInfrastructure();
+
+        constructProperty_scale_factors(SimTK::Vec3(1));
+        constructProperty_Appearance(Appearance());
+    }
+    // Default destructor
+    virtual ~Geometry() {}
+    /** Interface methods to handle the Frame which the Geometry is attached to.
+    **/
+    /** setFrameName sets the name of the Frame of attachment **/
+    void setFrameName(const std::string& name);
+    /** getFrameName returns a reference to the name of the Frame to which
+    this Geometry is attached (using a Connector). **/
+    const std::string& getFrameName() const;
+    /** getFrame returns a reference to the actual Frame to which this Geometry
+    is attached. */
+    const PhysicalFrame& getFrame() const;
+    //==========================================================================
+    // METHODS
+    //==========================================================================
+    // Compute Transform of this geometry relative to its base frame, utilizing 
+    // passed in state. Both transform and body_id are set in the passed-in 
+    // decorations as a side effect.
```

Comment:
Is this an internal class? If not shouldn't these all be doxygen comments rather than `//` comments?